### ISSUE TITLE
BUG: handleStrategyLogin is checking undefined property 'sync' without checking parent properties exist

### ISSUE THREAD

**Issue by mstrhakr** [Ylianst/MeshCentral#4520]
Posted on 25345fe by @softteh

> Hello! this change led to an error during authorization through MS Azure
> 
> ``` log
> TypeError: Cannot read property 'sync' of undefined
> at handleStrategyLogin (C:\Program Files\Open Source\MeshCentral\node_modules\meshcentral\webserver.js:2677:64)
> at Layer.handle [as handle_request] (C:\Program Files\Open Source\MeshCentral\node_modules\express\lib\router\layer.js:95:5)
> at next (C:\Program Files\Open Source\MeshCentral\node_modules\express\lib\router\route.js:144:13)
> at complete (C:\Program Files\Open Source\MeshCentral\node_modules\passport\lib\middleware\authenticate.js:271:13)
> at C:\Program Files\Open Source\MeshCentral\node_modules\passport\lib\middleware\authenticate.js:278:15
> at pass (C:\Program Files\Open Source\MeshCentral\node_modules\passport\lib\authenticator.js:428:14)
> at Authenticator.transformAuthInfo (C:\Program Files\Open Source\MeshCentral\node_modules\passport\lib\authenticator.js:450:5)
> at C:\Program Files\Open Source\MeshCentral\node_modules\passport\lib\middleware\authenticate.js:275:22
> at C:\Program Files\Open Source\MeshCentral\node_modules\passport\lib\http\request.js:41:7
> at C:\Program Files\Open Source\MeshCentral\node_modules\passport\lib\sessionmanager.js:26:5
> ```

This is a simple fix but can't be resolved until @Ylianst returns from vacation or has a free moment to merge #4521. I would recommend pinning to version [1.0.82](https://github.com/Ylianst/MeshCentral/releases/tag/1.0.82) or older until this is merged.

This line is the issue:
``` javascript
if (domain.authstrategies[authStrategy].groups.sync.enabled === true) { syncExternalUserGroups(domain, user, userMemberships, authStrategy) }
```
Wrapping it like this will check that the groups property exists as an object before checking further sub properties.
``` javascript
if (typeof domain.authstrategies[authStrategy].groups == 'object' || typeof domain.authstrategies[authStrategy].groups.sync == 'object') {
    if (domain.authstrategies[authStrategy].groups.sync.enabled === true) { syncExternalUserGroups(domain, user, userMemberships, authStrategy) }
}
```

IF you are having this issue, please roll back to [1.0.82](https://github.com/Ylianst/MeshCentral/releases/tag/1.0.82) or older as [1.0.83](https://github.com/Ylianst/MeshCentral/releases/tag/1.0.83) is where group sync was added

**Comment by mstrhakr** [comment]
Also #4517 That I didn't notice

**Comment by mstrhakr** [comment]
Sorry for the trouble @Ylianst, I promise to be good while you're gone from now on. :)

**Comment by zanderson-aim** [comment]
I also hit this with SAML, but you can also fix it by just adding the settings to the AuthStrategies section:

` "groups": { "sync": {  "enable": false  } }     `

Here is what my SAML looks like
```      
 "authStrategies": {
              "saml": {
                "newAccounts": true,
                "callbackurl": "https://...",
                "entityid": "...",
                "idpurl": "...",
                "cert": "saml.pem",
                "groups": { "sync": {  "enable": false  } }              
              },
```

**Comment by softteh** [comment]
thank you very much! Problem solved!

**Comment by mstrhakr** [comment]
> I also hit this with SAML, but you can also fix it by just adding the settings to the AuthStrategies section:
> 
> `"groups": { "sync": { "enable": false } } `
> 
> Here is what my SAML looks like
> 
> ```
>  "authStrategies": {
>               "saml": {
>                 "newAccounts": true,
>                 "callbackurl": "https://...",
>                 "entityid": "...",
>                 "idpurl": "...",
>                 "cert": "saml.pem",
>                 "groups": { "sync": {  "enable": false  } }              
>               },
> ```

This will no longer be necessary once you update as it will now check if the groups property exists as an object before checking if sync is enabled. 
This is a great workaround though if you prefer to wait for the upgrade vs downgrading.
Although I would add to it like this if you want to explicitly disable groups:
```
 "authStrategies": {
              "saml": {
                "newAccounts": true,
                "callbackurl": "https://...",
                "entityid": "...",
                "idpurl": "...",
                "cert": "saml.pem",
                "groups": { 
                    "enabled": false, // Add this line to ensure no other issues
                    "sync": {  
                        "enable": false
                    }
                 }              
              },
```

Just as a note, the groups functionality is only working with OIDC, although the groundwork is all laid for the other SSO types, just need help pulling groups from the other strategies.

**Comment by Ylianst** [comment]
Should be fixed in just released v1.0.85.
